src/cargo/util/toml/embedded.rs

```diff
@@ -229,16 +217,28 @@ fn split_source(input: &str) -> CargoResult<Source<'_>> {
         source.content = content;
     }
 
+    // Experiment: let us try which char works better
+    let tick_char = source
+        .content
+        .chars()
+        .filter(|c| ['`', '#', '-'].contains(c))
+        .next()
+        .unwrap_or('`');
+
     let tick_end = source
         .content
         .char_indices()
-        .find_map(|(i, c)| (c != '`').then_some(i))
+        .find_map(|(i, c)| (c != tick_char).then_some(i))
         .unwrap_or(source.content.len());
     let (fence_pattern, rest) = match tick_end {
         0 => {
             return Ok(source);
         }
         1 | 2 => {
+            if tick_char == '#' {
+                // Attribute
+                return Ok(source);
+            }
```

Comment:
Should this explicitly check for `#![]` or `#[]`?

Reply:
I'd rather not get into the complexity of matching rust syntax.  If we go this route, we can further evaluate what the exact behavior should be.